File: app/src/main/java/app/com/ezata/adapter/NewOrderAdapter.java
package app.com.ezata.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import app.com.ezata.R;
import app.com.ezata.model.NewOrder;
import app.com.ezata.ui.activities.MainActivity;
import app.com.ezata.ui.fragments.OrderDetailFragment;

public class NewOrderAdapter extends RecyclerView.Adapter<NewOrderAdapter.ViewHolder> {
    NewOrder[] orderList;
    Context context;

    public NewOrderAdapter(NewOrder[] orderList, Context context) {
        this.orderList = orderList;
        this.context=context;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View listItem= layoutInflater.inflate(R.layout.new_order_item, parent, false);
        ViewHolder viewHolder = new ViewHolder(listItem);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        NewOrder newOrder= orderList[position];
        holder.tvOrderNumber.setText(newOrder.getOrderNumber());
        holder.tvCustomerName.setText(newOrder.getCustomerName());
        holder.tvOrderTime.setText(newOrder.getOrderTime());
        holder.tvOrderItems.setText(newOrder.getOrderCount());
        holder.cardView.setOnClickListener(view -> {
        });

    }

    @Override
    public int getItemCount() {
        return orderList.length;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvOrderNumber, tvCustomerName, tvOrderTime, tvOrderItems;
        CardView cardView;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvOrderNumber=itemView.findViewById(R.id.tvOrderNumber);
            tvCustomerName=itemView.findViewById(R.id.tvCustomerName);
            tvOrderTime=itemView.findViewById(R.id.tvTime);
            tvOrderItems=itemView.findViewById(R.id.tvItemsCount);
            cardView=itemView.findViewById(R.id.cardview);
        }
    }
}

File: app/src/main/java/app/com/ezata/adapter/NewOrderAdapter.kt
package app.com.ezata.adapter

import android.content.Context
import app.com.ezata.model.NewOrder
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import app.com.ezata.R
import android.widget.TextView
import androidx.cardview.widget.CardView

class NewOrderAdapter(var orderList: Array<NewOrder>, var context: Context) :
    RecyclerView.Adapter<NewOrderAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val listItem =
            layoutInflater.inflate(R.layout.new_order_item, parent, false)
        return ViewHolder(listItem)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val newOrder = orderList[position]
        holder.tvOrderNumber.text = newOrder.orderNumber
        holder.tvCustomerName.text = newOrder.customerName
        holder.tvOrderTime.text = newOrder.orderTime
        holder.tvOrderItems.text = newOrder.orderCount
    }

    override fun getItemCount(): Int {
        return orderList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvOrderNumber: TextView = itemView.findViewById(R.id.tvOrderNumber)
        var tvCustomerName: TextView = itemView.findViewById(R.id.tvCustomerName)
        var tvOrderTime: TextView = itemView.findViewById(R.id.tvTime)
        var tvOrderItems: TextView = itemView.findViewById(R.id.tvItemsCount)
        var cardView: CardView = itemView.findViewById(R.id.cardview)

    }
}